contractcourt/commit_sweep_resolver_test.go

```diff
@@ -0,0 +1,178 @@
+package contractcourt
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/chainntnfs"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/sweep"
+)
+
+// TestCommitSweepResolverNoDelay tests resolution of a direct commitment output
+// unencumbered by a time lock.
+func TestCommitSweepResolverNoDelay(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	res := lnwallet.CommitOutputResolution{
+		SelfOutputSignDesc: input.SignDescriptor{
+			Output: &wire.TxOut{
+				Value: 100,
+			},
+		},
+	}
+
+	ctx := newCommitSweepResolverTestContext(t, &res)
+	ctx.resolve()
+
+	ctx.notifier.confChan <- &chainntnfs.TxConfirmation{}
+
+	// No csv delay, so the input should be swept immediately.
+	<-ctx.sweeper.sweptInputs
+
+	ctx.waitForResult()
+}
+
+// TestCommitSweepResolverDelay tests resolution of a direct commitment output
+// that is encumbered by a time lock.
+func TestCommitSweepResolverDelay(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	res := lnwallet.CommitOutputResolution{
+		SelfOutputSignDesc: input.SignDescriptor{
+			Output: &wire.TxOut{
+				Value: 100,
+			},
+		},
+		MaturityDelay: 3,
+	}
+
+	ctx := newCommitSweepResolverTestContext(t, &res)
+	ctx.resolve()
+
+	ctx.notifier.confChan <- &chainntnfs.TxConfirmation{
+		BlockHeight: testInitialBlockHeight - 1,
+	}
+
+	// Notify initial block height.
+	ctx.notifyEpoch(testInitialBlockHeight)
```

Comment:
should check that sweep is not attempted before the  maturity height

Reply:
added